packages/lib/modules/pool/pool.helpers.ts

```diff
@@ -514,12 +517,22 @@ export function allPoolTokens(pool: Pool | GqlPoolBase): TokenCore[] {
   return uniqBy(allTokens, 'address')
 }
 
-// Returns top level standard tokens + Erc4626 underlying tokens
+function shouldUseUnderlyingToken(token: PoolToken, pool: Pool | GqlPoolBase): boolean {
+  if (isV3Pool(pool) && token.isErc4626 && !token.underlyingToken) {
```

Comment:
v2 pools without `underlyingToken` would work but I prefer to be strict for now to detect potential API issues.